source/kraken/realtime.cpp

```diff
@@ -194,18 +220,32 @@ get_relationship(const transit_realtime::TripUpdate_StopTimeEvent& event,
 static nt::disruption::StopTimeUpdate::Status
 get_status(const transit_realtime::TripUpdate_StopTimeEvent& event,
            const transit_realtime::TripUpdate_StopTimeUpdate& st) {
-    auto transit_realtime_status = get_relationship(event, st);
-    switch (transit_realtime_status) {
-    case transit_realtime::TripUpdate_StopTimeUpdate_ScheduleRelationship_SKIPPED:
-        return nt::disruption::StopTimeUpdate::Status::DELETED;
-    case transit_realtime::TripUpdate_StopTimeUpdate_ScheduleRelationship_ADDED:
-        return nt::disruption::StopTimeUpdate::Status::ADDED;
-    default: break;
-    }
 
-    if (! event.has_delay() // for retrocompatibility since the old kirin version
-                                   // was not giving delays
-               || event.delay() != 0) {
+    if (event.HasExtension(kirin::stop_time_event_status)) {
+        const auto event_status = event.GetExtension(kirin::stop_time_event_status);
+        switch (event_status) {
+        case kirin::StopTimeEventStatus::ADDED:
+            return nt::disruption::StopTimeUpdate::Status::ADDED;
+        case kirin::StopTimeEventStatus::DELETED:
+            return nt::disruption::StopTimeUpdate::Status::DELETED;
+        case kirin::StopTimeEventStatus::DELETED_FOR_DETOUR:
+            return nt::disruption::StopTimeUpdate::Status::DELETED_FOR_DETOUR;
+        case kirin::StopTimeEventStatus::ADDED_FOR_DETOUR:
+            return nt::disruption::StopTimeUpdate::Status::ADDED_FOR_DETOUR;
+        default: break;
+        }
+    } else {
+        //TODO: to be deleted once this version is deployed in prod.
```

Comment:
`event.GetExtension(kirin::stop_time_event_status);` is new? if not, I think we can remove that, but I'm not sure.

Reply:
It's new and the else part will be deleted later